onnxruntime/core/providers/cpu/tensor/slice_helper.h

```diff
@@ -136,20 +136,12 @@ inline Status PrepareForComputeHelper(const gsl::span<const int64_t>& raw_starts
 
     // process end
     auto end = raw_ends[axis_index];
-    // INT_MAX has a special meaning for end according to spec
-    // equivalent to 'None' in numpy
-    // it represent slicing to the end of the dimension
-    if (end == std::numeric_limits<int32_t>::max() ||
-        end == std::numeric_limits<int64_t>::max()) {
-      end = step < 0 ? -1 : dim_value;
-    } else {
-      if (end < 0)
-        end += dim_value;
-      if (step < 0)
-        end = std::clamp(end, int64_t{-1}, dim_value);
-      else
-        end = std::clamp(end, int64_t{0}, dim_value);
-    }
+    if (end < 0)
+      end += dim_value;
+    if (step < 0)
+      end = std::clamp(end, int64_t{-1}, dim_value - 1);
```

Comment:
The change from dim_value -> dim_value - 1 may look arbitrary, but this is a spec clarification update as well. See https://github.com/onnx/onnx/pull/3908 and https://github.com/onnx/onnx/pull/4117 for details. 

You can't go wrong with either values ( `dim_values` or `dim_values-1` ) because if `ends` was clamped to either of these values while slicing in reverse, you will still have an empty output tensor, because one can't start from outside the array and reverse slice till the end and have a non-empty result. Using `dim_value - 1` as an `end` value is okay because the `end` value is exclusive. This change keeps it aligned to the "clarified" ONNX spec for Slice.